# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/core/platform.cpp
# Commit: 366030bd2d84
# Full Hash: 366030bd2d84770b99391ab5b3dee5b658ba9624
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2019-09-17 15:54:53
# Regressor Bug: 1576551
# File Overlap Count: 1
# Description:
#   Bug 1576551 - Assume capacity is in 8-byte entries - r=gregtatum
#   
#   Now that we are using a byte-oriented `BlocksRingBuffer` instead of an array of
#   9-byte `ProfileBufferEntry`'s, internally the profiler sets the buffer size in
#   bytes. However all external users (popup, tests, etc.) still assume that the
# ==============================================================================

diff -r d7ee4148aad9 -r 366030bd2d84 mozglue/baseprofiler/core/platform.cpp
--- a/mozglue/baseprofiler/core/platform.cpp	Tue Sep 17 01:49:59 2019 +0000
+++ b/mozglue/baseprofiler/core/platform.cpp	Tue Sep 17 01:50:19 2019 +0000
@@ -468,11 +468,11 @@
         mDuration(aDuration),
         mInterval(aInterval),
         mFeatures(AdjustFeatures(aFeatures, aFilterCount)),
-        mBuffer(MakeUnique<ProfileBuffer>(aCapacity))
+        // 8 bytes per entry.
+        mBuffer(MakeUnique<ProfileBuffer>(PowerOfTwo32(aCapacity.Value() * 8))),
         // The new sampler thread doesn't start sampling immediately because the
         // main loop within Run() is blocked until this function's caller
         // unlocks gPSMutex.
-        ,
         mSamplerThread(NewSamplerThread(aLock, mGeneration, aInterval))
 #  undef HAS_FEATURE
         ,
@@ -805,7 +805,7 @@
   const uint32_t mGeneration;
   static uint32_t sNextGeneration;
 
-  // The maximum number of entries in mBuffer.
+  // The maximum number of 8-byte entries in mBuffer.
   const PowerOfTwo32 mCapacity;
 
   // The maximum duration of entries in mBuffer, in seconds.
@@ -1737,7 +1737,7 @@
       "  first started.\n"
       "  If unset, the platform default is used:\n"
       "  %u entries per process, or %u when MOZ_BASE_PROFILER_STARTUP is set.\n"
-      "  (%zu bytes per entry -> %zu or %zu total bytes per process)\n"
+      "  (8 bytes per entry -> %u or %u total bytes per process)\n"
       "\n"
       "  MOZ_BASE_PROFILER_STARTUP_DURATION=<1..>\n"
       "  If MOZ_BASE_PROFILER_STARTUP is set, specifies the maximum life time\n"
@@ -1768,10 +1768,8 @@
       "default/unavailable)\n",
       unsigned(BASE_PROFILER_DEFAULT_ENTRIES.Value()),
       unsigned(BASE_PROFILER_DEFAULT_STARTUP_ENTRIES.Value()),
-      sizeof(ProfileBufferEntry),
-      sizeof(ProfileBufferEntry) * BASE_PROFILER_DEFAULT_ENTRIES.Value(),
-      sizeof(ProfileBufferEntry) *
-          BASE_PROFILER_DEFAULT_STARTUP_ENTRIES.Value());
+      unsigned(BASE_PROFILER_DEFAULT_ENTRIES.Value() * 8),
+      unsigned(BASE_PROFILER_DEFAULT_STARTUP_ENTRIES.Value() * 8));
 
 #  define PRINT_FEATURE(n_, str_, Name_, desc_)                             \
     printf("    %c %5u: \"%s\" (%s)\n",                                     \
@@ -2670,8 +2668,9 @@
 #  endif
 
   // Fall back to the default values if the passed-in values are unreasonable.
-  // Less than 8192 would not be enough for the most complex stack, so we should
-  // be able to store at least one full stack. TODO: Review magic numbers.
+  // Less than 8192 entries (65536 bytes) may not be enough for the most complex
+  // stack, so we should be able to store at least one full stack.
+  // TODO: Review magic numbers.
   PowerOfTwo32 capacity =
       (aCapacity.Value() >= 8192u) ? aCapacity : BASE_PROFILER_DEFAULT_ENTRIES;
   Maybe<double> duration = aDuration;